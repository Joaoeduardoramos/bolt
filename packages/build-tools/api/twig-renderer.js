const qs = require('querystring');
const fetch = require('node-fetch');
const config = require('@bolt/build-tools/utils/config-store').getConfig();
// const { getPort } = require('../utils/get-port');

async function twigRenderer(templatePath, body) {
  const options = {
    method: body ? 'POST' : 'GET',
  };
  if (body) options.body = JSON.stringify(body);
  const response = await fetch(`http://127.0.0.1:${config.renderingServicePort}?${qs.stringify({
    templatePath,
  })}`, options);
  const html = await response.text();

  const { status, statusText, ok } = response;
  return {
    status,
    statusText,
    ok,
    html,
  };
}

module.exports = {
  twigRenderer,
};
