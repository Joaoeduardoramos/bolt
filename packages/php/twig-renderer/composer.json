{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "description": "Twig rendering service used by the Bolt Design System.",
  "version": "2.8.0-beta.4",
  "license": "MIT",
  "repositories": [
	  {
      "type": "path",
	    "url": "../shared-twig-extensions"
	  },
    {
      "type": "path",
      "url": "../attributes-twig-extension"
    }
  ],
  "minimum-stability": "dev",
  "require": {
    "bolt-design-system/core-php": "*",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "1.37.1",
    "webmozart/path-util": "*",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
