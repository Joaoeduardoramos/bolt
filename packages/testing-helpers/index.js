export { isConnected } from './is-connected';
import {
  render as renderServer,
  renderString as renderStringServer,
  stop as stopServer,
} from '@bolt/twig-renderer';
export { fixture as html } from '@open-wc/testing-helpers';

export async function render(template, data) {
  return await renderServer(template, data, true);
}

export async function renderString(template, data) {
  return await renderStringServer(template, data, false);
}

export { stopServer };

// import {
//   // isConnected,
//   // render,
//   // renderString,
//   // stopServer,
//   html,
// } from '../../../testing-helpers';
