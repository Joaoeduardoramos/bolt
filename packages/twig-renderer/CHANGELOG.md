# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [2.3.0-rc.1](https://github.com/bolt-design-system/bolt/tree/master/packages/twig-renderer/compare/vv2.3.0-rc.0...v2.3.0-rc.1) (2019-01-04)

**Note:** Version bump only for package @bolt/twig-renderer





# [2.3.0-rc.0](https://github.com/bolt-design-system/bolt/tree/master/packages/twig-renderer/compare/v2.2.1...v2.3.0-rc.0) (2019-01-04)


### Bug Fixes

* re-enable schema validation for pattern lab but keep disabled for the docs site till 500 server error is troubleshooted ([ada76ef](https://github.com/bolt-design-system/bolt/tree/master/packages/twig-renderer/commit/ada76ef))
