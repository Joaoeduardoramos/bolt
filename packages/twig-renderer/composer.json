{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "version": "2.2.1",
  "license": "MIT",
  "minimum-stability": "dev",
  "require": {
    "twig/twig": "^1.0",
    "webmozart/path-util": "*",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
