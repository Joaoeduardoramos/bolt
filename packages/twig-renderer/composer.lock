{
    "_readme": [
        "This file locks the dependencies of your project to a known state",
        "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
        "This file is @generated automatically"
    ],
    "content-hash": "ad033ae6803f69e826643f9460e7080d",
    "packages": [
        {
            "name": "symfony/finder",
            "version": "dev-master",
            "source": {
                "type": "git",
                "url": "https://github.com/symfony/finder.git",
                "reference": "462d940115bc98dd81da28daf87ca7eaa8ce0552"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/symfony/finder/zipball/462d940115bc98dd81da28daf87ca7eaa8ce0552",
                "reference": "462d940115bc98dd81da28daf87ca7eaa8ce0552",
                "shasum": ""
            },
            "require": {
                "php": "^7.1.3"
            },
            "type": "library",
            "extra": {
                "branch-alias": {
                    "dev-master": "4.3-dev"
                }
            },
            "autoload": {
                "psr-4": {
                    "Symfony\\Component\\Finder\\": ""
                },
                "exclude-from-classmap": [
                    "/Tests/"
                ]
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "MIT"
            ],
            "authors": [
                {
                    "name": "Fabien Potencier",
                    "email": "fabien@symfony.com"
                },
                {
                    "name": "Symfony Community",
                    "homepage": "https://symfony.com/contributors"
                }
            ],
            "description": "Symfony Finder Component",
            "homepage": "https://symfony.com",
            "time": "2019-01-03T09:16:58+00:00"
        },
        {
            "name": "symfony/polyfill-ctype",
            "version": "v1.10.0",
            "source": {
                "type": "git",
                "url": "https://github.com/symfony/polyfill-ctype.git",
                "reference": "e3d826245268269cd66f8326bd8bc066687b4a19"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/symfony/polyfill-ctype/zipball/e3d826245268269cd66f8326bd8bc066687b4a19",
                "reference": "e3d826245268269cd66f8326bd8bc066687b4a19",
                "shasum": ""
            },
            "require": {
                "php": ">=5.3.3"
            },
            "suggest": {
                "ext-ctype": "For best performance"
            },
            "type": "library",
            "extra": {
                "branch-alias": {
                    "dev-master": "1.9-dev"
                }
            },
            "autoload": {
                "psr-4": {
                    "Symfony\\Polyfill\\Ctype\\": ""
                },
                "files": [
                    "bootstrap.php"
                ]
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "MIT"
            ],
            "authors": [
                {
                    "name": "Symfony Community",
                    "homepage": "https://symfony.com/contributors"
                },
                {
                    "name": "Gert de Pagter",
                    "email": "BackEndTea@gmail.com"
                }
            ],
            "description": "Symfony polyfill for ctype functions",
            "homepage": "https://symfony.com",
            "keywords": [
                "compatibility",
                "ctype",
                "polyfill",
                "portable"
            ],
            "time": "2018-08-06T14:22:27+00:00"
        },
        {
            "name": "twig/twig",
            "version": "1.x-dev",
            "source": {
                "type": "git",
                "url": "https://github.com/twigphp/Twig.git",
                "reference": "361d864178ea6abb86722633b6c9bd509a6e9930"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/twigphp/Twig/zipball/361d864178ea6abb86722633b6c9bd509a6e9930",
                "reference": "361d864178ea6abb86722633b6c9bd509a6e9930",
                "shasum": ""
            },
            "require": {
                "php": ">=5.4.0",
                "symfony/polyfill-ctype": "^1.8"
            },
            "require-dev": {
                "psr/container": "^1.0",
                "symfony/debug": "^2.7",
                "symfony/phpunit-bridge": "^3.4.19|^4.1.8"
            },
            "type": "library",
            "extra": {
                "branch-alias": {
                    "dev-master": "1.37-dev"
                }
            },
            "autoload": {
                "psr-0": {
                    "Twig_": "lib/"
                },
                "psr-4": {
                    "Twig\\": "src/"
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "BSD-3-Clause"
            ],
            "authors": [
                {
                    "name": "Fabien Potencier",
                    "email": "fabien@symfony.com",
                    "homepage": "http://fabien.potencier.org",
                    "role": "Lead Developer"
                },
                {
                    "name": "Armin Ronacher",
                    "email": "armin.ronacher@active-4.com",
                    "role": "Project Founder"
                },
                {
                    "name": "Twig Team",
                    "homepage": "https://twig.symfony.com/contributors",
                    "role": "Contributors"
                }
            ],
            "description": "Twig, the flexible, fast, and secure template language for PHP",
            "homepage": "https://twig.symfony.com",
            "keywords": [
                "templating"
            ],
            "time": "2019-01-14T09:42:01+00:00"
        },
        {
            "name": "webmozart/assert",
            "version": "1.4.0",
            "source": {
                "type": "git",
                "url": "https://github.com/webmozart/assert.git",
                "reference": "83e253c8e0be5b0257b881e1827274667c5c17a9"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/webmozart/assert/zipball/83e253c8e0be5b0257b881e1827274667c5c17a9",
                "reference": "83e253c8e0be5b0257b881e1827274667c5c17a9",
                "shasum": ""
            },
            "require": {
                "php": "^5.3.3 || ^7.0",
                "symfony/polyfill-ctype": "^1.8"
            },
            "require-dev": {
                "phpunit/phpunit": "^4.6",
                "sebastian/version": "^1.0.1"
            },
            "type": "library",
            "extra": {
                "branch-alias": {
                    "dev-master": "1.3-dev"
                }
            },
            "autoload": {
                "psr-4": {
                    "Webmozart\\Assert\\": "src/"
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "MIT"
            ],
            "authors": [
                {
                    "name": "Bernhard Schussek",
                    "email": "bschussek@gmail.com"
                }
            ],
            "description": "Assertions to validate method input/output with nice error messages.",
            "keywords": [
                "assert",
                "check",
                "validate"
            ],
            "time": "2018-12-25T11:19:39+00:00"
        },
        {
            "name": "webmozart/path-util",
            "version": "dev-master",
            "source": {
                "type": "git",
                "url": "https://github.com/webmozart/path-util.git",
                "reference": "95a8f7ad150c2a3773ff3c3d04f557a24c99cfd2"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/webmozart/path-util/zipball/95a8f7ad150c2a3773ff3c3d04f557a24c99cfd2",
                "reference": "95a8f7ad150c2a3773ff3c3d04f557a24c99cfd2",
                "shasum": ""
            },
            "require": {
                "php": "^5.3.3|^7.0",
                "webmozart/assert": "~1.0"
            },
            "require-dev": {
                "phpunit/phpunit": "^4.6",
                "sebastian/version": "^1.0.1"
            },
            "type": "library",
            "extra": {
                "branch-alias": {
                    "dev-master": "2.3-dev"
                }
            },
            "autoload": {
                "psr-4": {
                    "Webmozart\\PathUtil\\": "src/"
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "MIT"
            ],
            "authors": [
                {
                    "name": "Bernhard Schussek",
                    "email": "bschussek@gmail.com"
                }
            ],
            "description": "A robust cross-platform utility for normalizing, comparing and modifying file paths.",
            "time": "2016-08-15T15:31:42+00:00"
        }
    ],
    "packages-dev": [],
    "aliases": [],
    "minimum-stability": "dev",
    "stability-flags": [],
    "prefer-stable": false,
    "prefer-lowest": false,
    "platform": [],
    "platform-dev": []
}
