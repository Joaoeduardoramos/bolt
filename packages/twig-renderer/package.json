{
  "name": "@bolt/twig-renderer",
  "version": "2.2.1",
  "license": "MIT",
  "description": "Renders Twig",
  "main": "twig-renderer.js",
  "repository": "https://github.com/bolt-design-system/bolt/tree/master/packages/twig-renderer",
  "maintainers": [
    {
      "name": "Salem Ghoweri",
      "email": "me@salemghoweri.com",
      "web": "https://github.com/sghoweri"
    }
  ],
  "dependencies": {
    "@basalt/twig-renderer": "^0.10.2",
    "@bolt/build-tools": "^2.2.1",
    "sleep-promise": "^8.0.1"
  },
  "publishConfig": {
    "access": "public"
  },
  "schema": "utils/config.schema.yml",
  "bugs": "https://github.com/bolt-design-system/bolt/issues",
  "homepage": "https://bolt-design-system.com",
  "gitHead": "6368a1636ee39759167cd146de58a42fbd62453a"
}
