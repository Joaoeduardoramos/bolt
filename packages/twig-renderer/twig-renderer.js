const TwigRenderer = require('@basalt/twig-renderer');
const { getTwigNamespaceConfig } = require('@bolt/build-tools/utils/manifest');
const { getConfig } = require('@bolt/build-tools/utils/config-store');
const path = require('path');

let isReady = false;
let twigNamespaces;
let twigRenderer;

async function init() {
  const config = await getConfig();
  const relativeFrom = path.dirname(config.configFileUsed);
  // console.log({ config });
  twigNamespaces = await getTwigNamespaceConfig(
    relativeFrom,
    config.extraTwigNamespaces,
  );
  twigRenderer = new TwigRenderer({
    relativeFrom,
    src: {
      roots: [relativeFrom],
      namespaces: TwigRenderer.convertLegacyNamespacesConfig(twigNamespaces),
    },
    debug: true,
    alterTwigEnv: config.alterTwigEnv,
    hasExtraInfoInResponses: false, // Will add `info` onto results with a lot of info about Twig Env
  });
  isReady = true;
}

/**
 * Render Twig Template
 * @param {string} template - Template name (i.e. `@bolt/button.twig`)
 * @param {Object} data - Optional data to pass to template
 * @return {Promise<{{ ok: boolean, html: string, message: string}}>} - Results of render
 */
async function render(template, data = {}) {
  if (!isReady) {
    await init();
  }
  const results = await twigRenderer.render(template, data);
  // console.log({ results });
  return results;
}

/**
 * Render Twig String
 * @param {string} templateString - String that is a Twig template (i.e. `<p>{{ text }}</p>`)
 * @param {Object} data - Optional data to pass to template
 * @return {Promise<{{ ok: boolean, html: string, message: string}}>} - Results of render
 */
async function renderString(templateString, data = {}) {
  if (!isReady) {
    await init();
  }
  const results = await twigRenderer.renderString(templateString, data);
  // console.log({ results });
  return results;
}

// Both `render` and `renderString` will return something like this on success:
//   {
//     ok: true,
//     html: '<p>Hello World</p>',
//   }
// And something like this on failure:
//   {
//     ok: false,
//     message: 'The confabulator was not configured correctly',
//   }

module.exports = {
  render,
  renderString,
};
