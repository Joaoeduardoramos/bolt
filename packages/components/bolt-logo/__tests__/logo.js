import { render } from '@bolt/twig-renderer';
import { fixture as html } from '@open-wc/testing-helpers';

const timeout = 90000;

describe('logo', () => {
  let page;

  beforeEach(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
      waitLoad: true,
      waitNetworkIdle: true, // defaults to false
    });
  }, timeout);

  test('Basic usage', async () => {
    const results = await render('@bolt-components-logo/logo.twig', {
      src: '/fixtures/logo-paypal.svg',
      lazyload: false,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Logo with invert set to "true" renders properly', async () => {
    const results = await render('@bolt-components-logo/logo.twig', {
      src: '/fixtures/logo-paypal.svg',
      lazyload: false,
      invert: true,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Default <bolt-logo> renders', async function() {
    const renderedLogoHTML = await page.evaluate(() => {
      const logo = document.createElement('bolt-logo');

      // TODO: Use actual image web component once those changes are merged in
      logo.innerHTML = `<bolt-image>
                          <bolt-ratio ratio="124/33">
                            <img class="c-bolt-image__image"
                              srcset="/fixtures/logo-paypal.svg"
                              sizes="auto"
                              src="/fixtures/logo-paypal.svg">
                          </bolt-ratio>
                        </bolt-image>`;

      document.body.appendChild(logo);

      return logo.outerHTML;
    });

    // const renderedHTML = await html(renderedLogoHTML);
    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.01',
      failureThresholdType: 'percent',
    });

    // TODO: match snapshot once image is converted to web component, HTML is hardcoded here
    // expect(renderedHTML).toMatchSnapshot();
  });

  test('Inverted <bolt-logo> renders as inverted', async function() {
    const renderedLogoHTML = await page.evaluate(() => {
      const logo = document.createElement('bolt-logo');
      const wrapper = document.createElement('div');

      // TODO: Use actual image web component once those changes are merged in
      logo.innerHTML = `<bolt-image>
                          <bolt-ratio ratio="124/33">
                            <img class="c-bolt-image__image"
                              srcset="/fixtures/logo-paypal.svg"
                              sizes="auto"
                              src="/fixtures/logo-paypal.svg">
                          </bolt-ratio>
                        </bolt-image>`;

      logo.classList.add('c-bolt-logo--invert');

      wrapper.classList.add('t-bolt-dark');
      wrapper.appendChild(logo);

      document.body.appendChild(wrapper);

      return logo.outerHTML;
    });

    // const renderedHTML = await html(renderedLogoHTML);
    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.01',
      failureThresholdType: 'percent',
    });

    // TODO: match snapshot once image is converted to web component, HTML is hardcoded here
    // expect(renderedHTML).toMatchSnapshot();
  });
});
