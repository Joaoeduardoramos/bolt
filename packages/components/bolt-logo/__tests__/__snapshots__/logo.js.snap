// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logo Basic usage 1`] = `

<bolt-logo>
  <bolt-image>
    <bolt-ratio ratio="124/33">
      <replace-with-children class="c-bolt-ratio"
                             style="padding-bottom: calc((33 / 124) * 100%); --aspect-ratio: 3.7575757575758;"
      >
        <img class="c-bolt-image__image"
             srcset="/fixtures/logo-paypal.svg"
             sizes="auto"
             src="/fixtures/logo-paypal.svg"
        >
      </replace-with-children>
    </bolt-ratio>
  </bolt-image>
</bolt-logo>

`;

exports[`logo Logo with invert set to "true" renders properly 1`] = `

<bolt-logo class="c-bolt-logo--invert">
  <bolt-image>
    <bolt-ratio ratio="124/33">
      <replace-with-children class="c-bolt-ratio"
                             style="padding-bottom: calc((33 / 124) * 100%); --aspect-ratio: 3.7575757575758;"
      >
        <img class="c-bolt-image__image"
             srcset="/fixtures/logo-paypal.svg"
             sizes="auto"
             src="/fixtures/logo-paypal.svg"
        >
      </replace-with-children>
    </bolt-ratio>
  </bolt-image>
</bolt-logo>

`;
